exchanges/gateio/gateio_types.go

```diff
@@ -1793,12 +1793,19 @@ type DualModeResponse struct {
 	} `json:"history"`
 }
 
+// number represents a number type for JSON marshaling with zero value as "0"
```

Comment:
🥃 Hard to tell and I don't want to waste too much time, but I have a feeling that with encoding/json/v2 we can add a tag of `json:"price,format:customFormatQuoteZero"` and have our types.Number work generically for this. Obviously format would actually just be `quotezero` 😉 
I think when I sniffed this last time it was impossible with v2 at the field level because the field itself wouldn't get access to it's tags.
Now I'm thinking it might be possible by leveraging `inline` as well.

Just highlighting my thought train. I sniffed the new docs a bit on this but gave up as a poor use of time today.

Reply:
Good idea, we will probably have to remove sonic though when v2 is implemented which I am not mad about.